spec/models/manageiq/providers/openstack/cloud_manager/refresh_spec_common.rb

```diff
@@ -863,5 +863,107 @@ def expect_sync_cloud_tenants_with_tenants_is_queued
       expect(sync_cloud_tenant.method_name).to eq("sync_cloud_tenants_with_tenants")
       expect(sync_cloud_tenant.state).to eq(MiqQueue::STATE_READY)
     end
+
+    def assert_targeted_vm(vm_name, attributes)
```

Comment:
Oh, I think I see what you mean. "Targeted VM Refresh" just being the VM object itself, and "Full Targeted VM Refresh" being the VM and its relationships. Makes sense now. I'm refreshing the VM and its relationships, but it seems like not all the relevant information to populate the network stuff was available. (Since you only get the IP address not it's ID, and you don't have enough port information to reconstruct the whole network just from the VM and its relationships due to objects from the Openstack API not having references in both directions). Though if Neutron supports filtered querying like you mentioned, maybe I can get around that. I was working on this before 2 weeks of PTO so my memory is fuzzy about the exact nature of the problem.